docs/proposals/remediation/cluster-status-based-remedy.md

```diff
@@ -0,0 +1,383 @@
+	// If empty, the Actions will be performed immediately.
+	// +optional
+	DecisionMatches []DecisionMatch `json:"decisionMatches,omitempty"`
+
+	// Actions specifies the actions that remedy system needs to perform.
+	// If empty, no action will be performed.
+	// +optional
+	Actions []RemedyAction `json:"actions,omitempty"`
+}
+
+// DecisionMatch represents the decision match detail of activating the remedy system.
+type DecisionMatch struct {
+	// ClusterConditionMatch describes the cluster condition requirement.
+	// +optional
+	ClusterConditionMatch *ClusterConditionRequirement `json:"clusterConditionMatch,omitempty"`
+}
+
+// ClusterConditionRequirement describes the Cluster condition requirement details.
+type ClusterConditionRequirement struct {
+	// ConditionType specifies the ClusterStatus condition type.
+	ConditionType ConditionType `json:"conditionType"`
+	// Operator represents a conditionType's relationship to a conditionStatus.
+	// Valid operators are Equal, NotEqual.
+	Operator ClusterConditionOperator `json:"operator"`
+	// ConditionStatus specifies the ClusterStatue condition status.
+	ConditionStatus string `json:"conditionStatus"`
+}
+
+// ConditionType represents the detection ClusterStatus condition type.
+type ConditionType string
+
+const (
+	// ServiceDomainNameResolutionReady expresses the detection of the domain name resolution
+	// function of Service in the Kubernetes cluster.
+	ServiceDomainNameResolutionReady ConditionType = "ServiceDomainNameResolutionReady"
+)
+
+// ClusterConditionOperator is the set of operators that can be used in the cluster condition requirement.
+type ClusterConditionOperator string
+
+const (
+	// ClusterConditionEqual means equal match.
+	ClusterConditionEqual ClusterConditionOperator = "Equal"
+	// ClusterConditionNotEqual means not equal match.
+	ClusterConditionNotEqual ClusterConditionOperator = "NotEqual"
+)
+
+// ClusterAffinity represents the filter to select clusters.
+type ClusterAffinity struct {
+	// ClusterNames is the list of clusters to be selected.
+	// +optional
+	ClusterNames []string `json:"clusterNames,omitempty"`
+}
+
+// RemedyAction represents the action type the remedy system needs to preform.
+type RemedyAction string
+
+const (
+	// TrafficControl indicates that the cluster requires traffic control.
+	TrafficControl RemedyAction = "TrafficControl"
+)
+
+// RemedyList contains a list of Remedy.
+type RemedyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Remedy `json:"items"`
+}
+```
+
+The Cluster API definition is modified as follows:
+
+```go
+// ClusterStatus contains information about the current status of a
+// cluster updated periodically by cluster controller.
+type ClusterStatus struct {
+	...
+
+	// RemedyActions represents the remedy actions that needs to be performed
+	// on the cluster object.
+	// +optional
+    RemedyActions []string `json:"remedyActions,omitempty"`
```

Comment:
Do we need to record which `Remedy` a certain RemedyAction comes from?

Reply:
I understand that adding this information will be of great help to positioning problems. How to add this information specifically? I look forward to collecting more feedback from users, and then we will discuss and analyze whether to add it and how to add it.